pkg/controller/job/job_controller.go

```diff
@@ -1041,8 +1042,11 @@ func (jm *Controller) trackJobStatusAndRemoveFinalizers(ctx context.Context, job
 			break
 		}
 	}
-	if len(newSucceededIndexes) > 0 {
+	if isIndexed {
 		succeededIndexes = succeededIndexes.withOrderedIndexes(newSucceededIndexes)
+		if succeededIndexes.String() != job.Status.CompletedIndexes {
+			needsFlush = true
+		}
 		job.Status.Succeeded = int32(succeededIndexes.total())
 		job.Status.CompletedIndexes = succeededIndexes.String()
```

Comment:
we are calling `String()` twice, which is not very expensive, but not trivial and can be avoided. Save it in variable above.

Reply:
Done